src/imarker_robot_state.cpp

```diff
@@ -108,39 +114,40 @@ bool isIKStateValid(const planning_scene::PlanningScene* planning_scene, bool ve
 
 namespace moveit_visual_tools
 {
-IMarkerRobotState::IMarkerRobotState(planning_scene_monitor::PlanningSceneMonitorPtr psm,
-                                     const std::string& imarker_name, std::vector<ArmData> arm_datas,
-                                     rviz_visual_tools::colors color, const std::string& package_path)
+IMarkerRobotState::IMarkerRobotState(rclcpp::Node::SharedPtr node, planning_scene_monitor::PlanningSceneMonitorPtr psm, const std::string& imarker_name,
+                    std::vector<ArmData> arm_datas, rviz_visual_tools::Colors color, const std::string& package_path)
   : name_(imarker_name)
-  , nh_("~")
   , arm_datas_(std::move(arm_datas))
   , psm_(std::move(psm))
   , color_(color)
   , package_path_(package_path)
 {
   // Load Visual tools with respect to Eigen memory alignment
+  // ! Add node here
   visual_tools_ = std::allocate_shared<moveit_visual_tools::MoveItVisualTools>(
-      Eigen::aligned_allocator<moveit_visual_tools::MoveItVisualTools>(), psm_->getRobotModel()->getModelFrame(),
-      nh_.getNamespace() + "/" + imarker_name, psm_);
+      Eigen::aligned_allocator<moveit_visual_tools::MoveItVisualTools>(), node, psm_->getRobotModel()->getModelFrame(), "/" + imarker_name, psm_);
 
   // visual_tools_->setPlanningSceneMonitor(psm_);
-  visual_tools_->loadRobotStatePub(nh_.getNamespace() + "/imarker_" + imarker_name + "_state");
+  visual_tools_->loadRobotStatePub("/imarker_" + imarker_name + "_state");
 
   // Load robot state
   imarker_state_ = std::make_shared<moveit::core::RobotState>(psm_->getRobotModel());
   imarker_state_->setToDefaultValues();
 
   // Create Marker Server
-  const std::string imarker_topic = nh_.getNamespace() + "/" + imarker_name + "_imarker";
-  imarker_server_ = std::make_shared<interactive_markers::InteractiveMarkerServer>(imarker_topic, "", false);
+  const std::string imarker_topic = "/" + imarker_name + "_imarker";
+  // ! Add node here
+  // imarker_server_ = std::make_shared<interactive_markers::InteractiveMarkerServer>(imarker_topic, "", false);
+  // TODO: Check topic ns or imarker_topic
+  imarker_server_ = std::make_shared<interactive_markers::InteractiveMarkerServer>("", node);
```

Comment:
Seems like the `imarker_topic` is not applied?

Reply:
So the topic namespace should still be used, the [constructor](https://github.com/ros-visualization/interactive_markers/blob/ros2/src/interactive_marker_server.cpp#L48) accepts that as well. I think it's still pretty important that namespaces are used everywhere, also in line 130 above.